compiler/src/iree/compiler/Codegen/LLVMCPU/Utils.cpp

```diff
@@ -58,4 +58,44 @@ bool hasI8mmFeature(IREE::HAL::ExecutableTargetAttr targetAttr) {
   return hasFeature(targetAttr, "+i8mm");
 }
 
+bool transposeLoweringPreconditionAArch64SME(linalg::GenericOp genericOp) {
+  // Check op has 2 dimensions.
+  if (genericOp.getNumLoops() != 2)
+    return false;
+
+  // Check op has single input and output.
+  if ((genericOp.getNumDpsInputs() != 1) || (genericOp.getNumDpsInits() != 1))
+    return false;
```

Comment:
nit: we don't need `()` for the two conditions.

```suggestion
  if (genericOp.getNumDpsInputs() != 1 || genericOp.getNumDpsInits() != 1)
    return false;
```